ssh/tailssh/tailssh.go

```diff
@@ -337,6 +338,23 @@ func (c *conn) fakePasswordHandler(ctx ssh.Context, password string) bool {
 	return c.anyPasswordIsOkay
 }
 
+// publicKeyHandler is our implementation of the PublicKeyHandler hook that
+// checks whether the user's public key is correct. It exists for clients that
+// don't support "none" auth and instead insist on supplying a public key.
+// This ignores the supplied public key and authenticates with Tailscale auth
+// in the same way as NoClientAuthCallback.
+func (c *conn) publicKeyHandler(ctx ssh.Context, pubKey ssh.PublicKey) error {
+	if err := c.doPolicyAuth(ctx); err != nil {
+		c.logf("rejecting SSH public key %s: %v", bytes.TrimSpace(gossh.MarshalAuthorizedKey(pubKey)), err)
+		return err
+	}
+	if err := c.isAuthorized(ctx); err != nil {
+		return err
+	}
+	c.logf("accepting SSH public key %s", bytes.TrimSpace(gossh.MarshalAuthorizedKey(pubKey)))
```

Comment:
let's make this message a bit less misleading. It kinda suggests we care/know about that key

Reply:
and do we even care to log the MarshalAuthorizedKey value? 

Maybe "confused client presented an unnecessary SSH public key; pacifying already-authenticated client by saying it's fine"

Reply:
I'll just take it out. We don't log results in the `NoClientAuthCallback` and I don't know that we need to here either.